### PR TITLE
I23 grid scans

### LINKED ISSUE #574: I23: Do a stepwise gridscan with I23 motors
To do the xray centre on i23 we will need a stepwise gridscan.

## Acceptance Criteria
* There are simulated S23 motors
* The motors on I23 can be moved in the expected pattern (horizontal snaked)

## Note
* Don't need to do anything with detectors etc. this is just motion
* Should be able to use [scan spec](https://github.com/dls-controls/scanspec) for this

### LINKED ISSUE #578: I23: Create an endpoint for the xray centre
We could potentially reuse the one i03 uses, will need to think about it

### PR DESCRIPTION
**Pull request by NeilSmithDLS**
Fixes #578 and fixes #574

Link to dodal PR (if required): #59 (https://github.com/DiamondLightSource/dodal/pull/59)

This change introduces the stepped_grid_scan end point and the associated changes required to perform the basic movement required for a stepped grid scan.

### To test:
1. Ensure https://gerrit.diamond.ac.uk/c/gda/gda-mx/+/38922 is installed into GDA.
2. Run GDA for i23
3. Press the "Xray Centring" button on the OAV controls area
4. Within Hyperion confirm that the stepped grid scan plan is called and the x & y motors are moved

### CODE REVIEW

File: gridscan_playing.py
Comment (NeilSmithDLS) [inline gridscan_playing.py:1]: This file si not needed and should be deleted from the Pull Request

File: src/artemis/experiment_plans/stepped_grid_scan_plan.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    The ispyb handler should be added to the whole gridscan as we want to capture errors
+    at any point in it.
+
+    Args:
+        parameters (SteppedGridScanInternalParameters): The parameters to run the scan.
+
+    Returns:
+        Generator: The plan for the gridscan
+    """
+    LOGGER.info("******* get_plan called")
```
Comment (NeilSmithDLS) [inline src/artemis/experiment_plans/stepped_grid_scan_plan.py:152]: This line should be removed

File: gridscan_playing.py
Comment (NeilSmithDLS) [reply gridscan_playing.py:1]: file removed

File: src/artemis/experiment_plans/stepped_grid_scan_plan.py (hunk 1 above)
Comment (NeilSmithDLS) [reply src/artemis/experiment_plans/stepped_grid_scan_plan.py:152]: done

File: src/artemis/device_setup_plans/setup_zebra_for_stepped_grid_scan.py
[the remark this answers is not in the document]
Comment (NeilSmithDLS) [reply src/artemis/device_setup_plans/setup_zebra_for_stepped_grid_scan.py:1]: file removed

File: src/artemis/experiment_plans/stepped_grid_scan_plan.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    from gridscan processing results)"""
+    LOGGER.info(f"Moving Smargon x, y, z to: {xray_centre_motor_position}")
+    yield from bps.mv(
+        sample_motors.x,
+        xray_centre_motor_position.x,
+        sample_motors.y,
+        xray_centre_motor_position.y,
+        sample_motors.z,
+        xray_centre_motor_position.z,
+    )
```
Comment (DominicOram) [inline src/artemis/experiment_plans/stepped_grid_scan_plan.py:89]: Should: Maybe a new issue but this is an exact copy of the `fast_grid_scan` one, it should probably be somewhere common

**Review by DominicOram — CHANGES_REQUESTED** [review]
Really good basis to start from, thank you! Some minor comments in code of places we can remove a bit more that was copied over

**Comment by DominicOram** [comment]
Sorry, just realised this is on me. Will  review now

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**